frontend/src/components/grant/tabs/Visuals.tsx

```diff
@@ -8,28 +14,75 @@ interface GrantMainTabProps {
 
 const Visuals:React.FC<GrantMainTabProps> = ({grantData}) => {
 
+  const pdfRef = useRef(null);
+
+  const [buttonVisible, setButtonVisible] = useState(true);
+
+  const downloadPDF = () =>{
+    const input = pdfRef.current;
+
+    if (input){
+
+      setButtonVisible(false);
+
+      html2canvas(input).then((canvas) =>{
+
+        setButtonVisible(true);
+
+        const imgData =  canvas.toDataURL('image/png');
+        const pdf = new jsPDF('p','mm','a4',true);
+        const pdfWidth = pdf.internal.pageSize.getWidth();
+        const pdfHeight = pdf.internal.pageSize.getHeight();
+        const imgWidth = canvas.width;
+        const imgHeight = canvas.height;
+        const ratio = Math.min(pdfWidth / imgWidth, pdfHeight / imgHeight);
+        const imgX = (pdfWidth - imgWidth * ratio) / 2;
+        const imgY = 30;
+
+        pdf.addImage(imgData,'PNG',imgX,imgY,imgWidth * ratio,imgHeight * ratio);
+        pdf.save('charts.pdf');
+        
+
+      });
+    }
+  };
+
   return(
-    <><div>
-      Visuals tab content
-    </div>
-    <div className="flex flex-row">
+    <>
+      <div className="flex flex-row justify-center pt-4">
 
-      <div>
+        <div>
       Grant Start Date
-        <CalendarView displayDate={grantData.startDate} />
-      </div>
+          <CalendarView displayDate={grantData.startDate} />
+        </div>
 
-      <div className="px-4">
+        <div className="px-4">
       Today's Date
-        <CalendarView displayDate={new Date()} />
-      </div>
+          <CalendarView displayDate={new Date()} />
+        </div>
 
-      <div>
+        <div>
       Grant End Date
-        <CalendarView displayDate={grantData.endDate} />
+          <CalendarView displayDate={grantData.endDate} />
+        </div>
+
       </div>
+      
+      <div className="flex flex-row w-full justify-center" ref = {pdfRef}>
 
-    </div>
+        <div className="flex flex-row justify-between items-center p-4">
+          <GrantBarChart />
+        </div>
+
+        <div className="flex flex-row justify-between items-center p-4">
+          <GrantLineChart />
+        </div>
+        
+      </div>
```

Comment:
not gonna boot out the PR for this but seems like indentation is janky -- i'll fix it later